### PR TITLE
Radix sort stability docs

### LINKED ISSUE #282: Document that cub::DeviceRadixSort and cub::BlockRadixSort are stable
There is nothing in the documentation https://nvlabs.github.io/cub/structcub_1_1_device_radix_sort.html saying whether it is stable sort or not. It seems to be stable. If so, it would be great if this fact is mentioned in the documentation.

### PR DESCRIPTION
**Pull request by allisonvacanti**
Fixes #282, extends and replaces #284.

### CODE REVIEW

File: cub/device/device_radix_sort.cuh [hunk 1]
```diff
@@ -66,9 +66,9 @@ namespace cub {
  * half-precision floating-point type.  Although the direct radix sorting
  * method can only be applied to unsigned integral types, DeviceRadixSort
  * is able to sort signed and floating-point types via simple bit-wise transformations
- * that ensure lexicographic key ordering. For floating-point types -0.0 and +0.0 are
- * considered equal and appear in the result in the same order as they appear in
- * the input.
+ * that ensure lexicographic key ordering. DeviceRadixSort is stable. For
+ * floating-point types -0.0 and +0.0 are considered equal and appear in the
+ * result in the same order as they appear in the input.
```
Comment (zasdfgbnm) [inline cub/device/device_radix_sort.cuh:71]: Can we also document that cub does not support in-place operations, so input and output pointers can not have memory overlap?

Comment (allisonvacanti) [reply cub/device/device_radix_sort.cuh:71]: Absolutely, I'll add that before this goes in. 👍

**Comment by allisonvacanti** [comment]
@zasdfgbnm I added some more documentation to the sort algorithms in the last push, including #293. Can you review these changes to make sure they provide enough information for your needs?

File: cub/block/block_radix_sort.cuh [hunk 2]
```diff
@@ -64,23 +64,53 @@ namespace cub {
...
+ * given input sequence of keys and a set of rules specifying a total ordering
+ * of the symbolic alphabet, the radix sorting method produces a lexicographic
+ * ordering of those keys.
+ *
+ * \rowmajor
+ *
+ * \par Supported Types
+ * BlockRadixSort can sort all of the built-in C++ numeric primitive types
+ * (<tt>unsigned char</tt>, \p int, \p double, etc.) as well as CUDA's \p __half
+ * half-precision floating-point type.
```
Comment (zasdfgbnm) [inline cub/block/block_radix_sort.cuh:80]: this might need to add `__nv_bfloat16`

File: cub/block/block_radix_sort.cuh [hunk 3]
```diff
@@ -64,23 +64,53 @@ namespace cub {
...
+ *
+ * \par No Descending Sort Transformations
+ * Unlike `DeviceRadixSort`, `BlockRadixSort` does not invert the input key bits
+ * when performing a descending sort. Instead, it has special logic to reverse
+ * the order of the keys while sorting.
+ *
+ * \par Stability
+ * BlockRadixSort is stable. For floating-point types -0.0 and +0.0
+ * are considered equal and appear in the result in the same order as they
+ * appear in the input.
```
Comment (zasdfgbnm) [inline cub/block/block_radix_sort.cuh:112]: How about adding something about the behavior of nan?

File: cub/device/device_segmented_radix_sort.cuh [hunk 4]
```diff
@@ -66,9 +66,9 @@ namespace cub {
  * half-precision floating-point type.  Although the direct radix sorting
  * method can only be applied to unsigned integral types, DeviceSegmentedRadixSort
  * is able to sort signed and floating-point types via simple bit-wise transformations
- * that ensure lexicographic key ordering. For floating-point types -0.0 and +0.0 are
- * considered equal and appear in the result in the same order as they appear in
- * the input.
+ * that ensure lexicographic key ordering. DeviceSegmentedRadixSort is stable. For
+ * floating-point types -0.0 and +0.0 are considered equal and appear in the
+ * result in the same order as they appear in the input.
```
Comment (zasdfgbnm) [inline cub/device/device_segmented_radix_sort.cuh:71]: Same here:
How about adding something about the behavior of nan?

File: cub/block/block_radix_sort.cuh (hunk 2 above)
Comment (allisonvacanti) [reply cub/block/block_radix_sort.cuh:80]: I have updated #306 to mention bfloat16. I'll update that PR after this goes in.

File: cub/block/block_radix_sort.cuh (hunk 3 above)
Comment (allisonvacanti) [reply cub/block/block_radix_sort.cuh:112]: 👍🏻

**Pull request merged**